src/UserRepository/PdoDatabase.php

```diff
@@ -64,20 +71,32 @@ public function authenticate(string $credential, string $password = null) : ?Use
     /**
      * {@inheritDoc}
      */
-    public function getRolesFromUser(string $username) : array
+    public function getRolesFromUser(string $identity) : array
     {
         if (! isset($this->config['sql_get_roles'])) {
             return [];
         }
 
-        if (false === strpos($this->config['sql_get_roles'], ':username')) {
+        if (false === strpos($this->config['sql_get_roles'], ':identity')) {
             throw new Exception\InvalidConfigException(
-                'The sql_get_roles configuration setting must include a :username parameter'
+                'The sql_get_roles configuration setting must include a :identity parameter'
             );
         }
 
-        $stmt = $this->pdo->prepare($this->config['sql_get_roles']);
-        $stmt->bindParam(':username', $username);
+        try {
+            $stmt = $this->pdo->prepare($this->config['sql_get_roles']);
+        } catch (PDOException $e) {
+            throw new Exception\RuntimeException(sprintf(
+                "Error on sql_get_rols configuration: %s",
+                $e->getMessage()
+            ));
+        }
+        if (false === $stmt) {
+            throw new Exception\RuntimeException(
+                "Error on sql_get_rols configuration"
+            );
```

Comment:
Emit exactly the same exception message here as above, as this particular conditional is testing the same thing (error with statement preparation), but with PDO exceptions disabled. The only change is that the `%s` would become `unknown error`.

Reply:
So maybe it should be moved into `Exception\RuntimeException` class with static method and parameter for `%s` as we already have in other expressive components?